python/dgl/data/wiki_network.py

```diff
@@ -0,0 +1,186 @@
+    Embedding <https://arxiv.org/abs/1909.13021>`__
+
+    Parameters
+    ----------
+    name : str
+        Name of the dataset.
+    raw_dir : str
+        Raw file directory to store the processed data.
+    force_reload : bool
+        Whether to always generate the data from scratch rather than load a
+        cached version.
+    verbose : bool
+        Whether to print progress information.
+    transform : callable
+        A transform that takes in a :class:`~dgl.DGLGraph` object and returns
+        a transformed version. The :class:`~dgl.DGLGraph` object will be
+        transformed before every access.
+    """
+
+    def __init__(self, name, raw_dir, force_reload, verbose, transform):
+        url = _get_dgl_url(f"dataset/{name}.zip")
+        super(WikiNetworkDataset, self).__init__(
+            name=name,
+            url=url,
+            raw_dir=raw_dir,
+            hash_key=(url),
+            force_reload=force_reload,
+            verbose=verbose,
+            transform=transform,
+        )
+
+    def process(self):
+        """Load and process the data."""
+        # node features and labels
+        with open(f"{self.raw_path}/out1_node_feature_label.txt", "r") as f:
+            data = f.read().split("\n")[1:-1]
+        features = [
+            [float(v) for v in r.split("\t")[1].split(",")] for r in data
+        ]
+        features = F.tensor(features, dtype=F.data_type_dict["float32"])
+        labels = [int(r.split("\t")[2]) for r in data]
+        self._num_classes = max(labels) + 1
+        labels = F.tensor(labels, dtype=F.data_type_dict["int64"])
+
+        # graph structure
+        with open(f"{self.raw_path}/out1_graph_edges.txt", "r") as f:
+            data = f.read().split("\n")[1:-1]
+            data = [[int(v) for v in r.split("\t")] for r in data]
+        COO = F.swapaxes(F.tensor(data, dtype=F.data_type_dict["int64"]), 0, 1)
+        dst = F.squeeze(F.narrow_row(COO, start=0, stop=1), dim=0)
+        src = F.squeeze(F.narrow_row(COO, start=1, stop=2), dim=0)
+
+        self._g = graph((src, dst), num_nodes=F.shape(features)[0])
+        self._g.ndata["feat"] = features
+        self._g.ndata["label"] = labels
+
+        # split
+        train_masks, val_masks, test_masks = [], [], []
+        for i in range(10):
+            filepath = f"{self.raw_path}/{self.name}_split_0.6_0.2_{i}.npz"
+            f = np.load(filepath)
+            train_masks += [F.zerocopy_from_numpy(f["train_mask"])]
+            val_masks += [F.zerocopy_from_numpy(f["val_mask"])]
+            test_masks += [F.zerocopy_from_numpy(f["test_mask"])]
+        self._g.ndata["train_mask"] = F.astype(
+            F.stack(train_masks, dim=1), F.data_type_dict["bool"]
+        )
+        self._g.ndata["val_mask"] = F.astype(
+            F.stack(val_masks, dim=1), F.data_type_dict["bool"]
+        )
+        self._g.ndata["test_mask"] = F.astype(
+            F.stack(test_masks, dim=1), F.data_type_dict["bool"]
+        )
```

Comment:
The general question is whether the logic of data processing / graph construction should belong to DGL core codebase or not.

Reply:
Upon the offline discussion, the general principle will be including the logic of data processing in the core DGL codebase.